src/wasm/wasm-emscripten.cpp

```diff
@@ -739,6 +739,43 @@ void printSignatures(std::ostream& o, const std::set<Signature>& c) {
   o << "]";
 }
 
+static bool mainReadsParams(Module& module) {
+  // In the new llvm name mangling scheme for `main` we get the
+  // 2 argument main mangled as `__main_argc_argv` and the no
+  // argument form generates and alias called `__main_void`
+  if (module.getExportOrNull("__main_void")) {
```

Comment:
I don't follow... what is an alias for what? (`main_void` can't be an alias for `main_argc_argv` I don't think as they have different types, so must be something else?)

Reply:
If your main function has two args it gets mangled to `__main_argc_argv`.

If your main function has zero args then it does not get mangled, but an alias to it created called `__main_void`.

Therefore if `__main_void` exists then we know for sure that main does *not* read params.